codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithygo/SymbolVisitor.java

```diff
@@ -0,0 +1,518 @@
+        return StringUtils.capitalize(removeLeadingInvalidIdentCharacters(shape.getId().getName(serviceShape)));
+    }
+
+    private String getDefaultMemberName(MemberShape shape) {
+        String memberName = StringUtils.capitalize(removeLeadingInvalidIdentCharacters(shape.getMemberName()));
+
+        return memberName;
+    }
+
+    private String removeLeadingInvalidIdentCharacters(String value) {
+        if (Character.isAlphabetic(value.charAt(0))) {
+            return value;
+        }
+
+        int i;
+        for (i = 0; i < value.length(); i++) {
+            if (Character.isAlphabetic(value.charAt(i))) {
+                break;
+            }
+        }
+
+        String remaining = value.substring(i);
+        if (remaining.length() == 0) {
+            throw new CodegenException("tried to clean name " + value + ", but resulted in empty string");
+        }
+
+        return remaining;
+    }
+
+
+    private boolean isErrorMember(MemberShape shape) {
+        return model.getShape(shape.getContainer())
+                .map(container -> container.hasTrait(ErrorTrait.ID))
+                .orElse(false);
+    }
+
+    @Override
+    public Symbol blobShape(BlobShape shape) {
+        if (shape.hasTrait(StreamingTrait.ID)) {
+            Symbol inputVariant = symbolBuilderFor(shape, "Reader", SmithyGoDependency.IO).build();
+            return symbolBuilderFor(shape, "ReadCloser", SmithyGoDependency.IO)
+                    .putProperty(SymbolUtils.INPUT_VARIANT, inputVariant)
+                    .build();
+        }
+        return symbolBuilderFor(shape, "[]byte")
+                .putProperty(SymbolUtils.GO_UNIVERSE_TYPE, true)
+                .build();
+    }
+
+    @Override
+    public Symbol booleanShape(BooleanShape shape) {
+        return symbolBuilderFor(shape, "bool").definitionFile("./TypeConversion.go").namespace("typeconversion", "")
+                .putProperty(SymbolUtils.GO_UNIVERSE_TYPE, true)
+                .putProperty("native", symbolBuilderFor(shape, "bool").definitionFile("./TypeConversion.go").build())
+                .putProperty("dafny", symbolBuilderFor(shape, "Wrappers_Compile.Option").addDependency("Wrappers_Compile", "").definitionFile("./TypeConversion.go").build())
+                       .putProperty("ToNative", "val.Dtor_value().(bool)")
+                                              .putProperty("ToDafny", "Wrappers_Compile.Companion_Option_.Create_Some_(val)")
+
+                                              .build();
+    }
+
+    @Override
+    public Symbol listShape(ListShape shape) {
+        return createCollectionSymbol(shape);
+    }
+
+    @Override
+    public Symbol setShape(SetShape shape) {
+        // Go doesn't have a set type. Rather than hack together a set using a map,
+        // we instead just create a list and let the service be responsible for
+        // asserting that there are no duplicates.
+        return createCollectionSymbol(shape);
+    }
```

Comment:
Issue: Hopefully, this never comes up.
But this will not work for a local service...
If Dafny expects a Set, it will need to be a Set.

For now, I would either add a TODO or log a warning.

Reply:
This shouldn't be an issue, since we anyway need to handle this in the `TypeConversion`.